app/src/main/java/com/sunnysouth/view/ui/activity/RegisterActivity.kt

```diff
@@ -3,133 +3,164 @@ package com.sunnysouth.view.ui.activity
         super.onCreate(savedInstanceState)
         //setContentView
+        viewModel.setContextApp(this)
         setContentView(R.layout.activity_register)
 
-        //InvisibleButtons
-        val formp4 = findViewById(R.id.go_to_register_form_p4) as? MaterialButton
-        formp4?.visibility = View.INVISIBLE
+        this.passWordDataBtn = (findViewById(R.id.passwordBtn) as? MaterialButton)!!
+        this.credentialDataBtn = (findViewById(R.id.credentialBtn) as? MaterialButton)!!
+        this.personalDataBtn = (findViewById(R.id.personalBtn) as? MaterialButton)!!
 
-        val formp3 = findViewById(R.id.go_to_register_form_p3) as? MaterialButton
-        formp3?.visibility = View.INVISIBLE
+        passWordDataBtn?.visibility = View.INVISIBLE
+        personalDataBtn?.visibility = View.VISIBLE
+        credentialDataBtn?.visibility = View.INVISIBLE
 
-        val fragment_personal_data = PersonalDataFragment.newInstance()
-        replaceFragment(fragment_personal_data,"personalFragment")
-
-        val formp2 = findViewById(R.id.go_to_register_form_p2) as? MaterialButton
-
-        var user = User()
+        replaceFragment(fragment_personal_data, "personalFragment")
 
         //Click Envent 'Siguiente#1'
-        formp2?.setOnClickListener{
+        personalDataBtn?.setOnClickListener{
             //get the text
-            user.firstName = first_name?.editText?.text.toString()
-
-            /*first_name.addOnEditTextAttachedListener {
-                // If any specific changes should be done when the edit text is attached (and
-                // thus when the trailing icon is added to it), set an
-                // OnEditTextAttachedListener.
-
-                // Example: The clear text icon's visibility behavior depends on whether the
-                // EditText has input present. Therefore, an OnEditTextAttachedListener is set
-                // so things like editText.getText() can be called.
-            }
-            */
-            user.lastName = last_name?.editText?.text.toString()
-            user.phoneNumber = phone_number?.editText?.text.toString()
-
-            //Toast.makeText(this,user.first_name,Toast.LENGTH_LONG).show()
-
-            if(user.personalDataIsValid())
-            {
-                if(user.phoneNumberValidation())
-                {
-                    formp3?.visibility = View.VISIBLE
-                    formp2?.visibility = View.INVISIBLE
+            var firstName = findViewById<EditText>(R.id.first_name)
+            var lastName = findViewById<EditText>(R.id.last_name)
+            var phoneNumber = findViewById<EditText>(R.id.phone_number)
+
+            viewModel?.setUserPersonalData(firstName?.text.toString(), lastName?.text.toString(), phoneNumber?.text.toString())
+
+            viewModel.user.observe(
+                this,
+                Observer<User> {
+                    var userObj = it
+                    if(it.IsValidpersonalData())
+                    {
```

Comment:
el code style de kotlin es con los braquets en la misma linea del metodo `if(it.IsValidpersonalData()){`